### DISCUSSION TITLE
Does the mobile app support proxy/socks5?

### DISCUSSION THREAD

**Discussion by go-vegan** [immich-app/immich#5360] [Feature Request]
### The feature

Does the mobile app support proxy/socks5?
If so, how can it be used?
I didn't find such option in the settings.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by gadore** [comment]
I test it out, for my "Server not reachable" case, it's mobile proxy setting issue.

Immich doesn't read and use mobile device's proxy setting. I have tested it out, after add proxy overide, it won't show "Server not reachable" again.

See code below:
```
import 'package:http_proxy/http_proxy.dart';
// ……
void main() async {
  ImmichWidgetsBinding();
  final db = await loadDb();
  await initApp();
  await migrateDatabaseIfNeeded(db);
  HttpOverrides.global = HttpSSLCertOverride();

  // ######## NEW CODE AREA START
  // Test Only, need discussion and with more elegant method to fullfill it.
  WidgetsFlutterBinding.ensureInitialized();
  HttpProxy httpProxy = await HttpProxy.createHttpProxy();
  HttpOverrides.global = httpProxy;
  // ######## NEW CODE AREA END

  runApp(
    ProviderScope(
      overrides: [dbProvider.overrideWithValue(db)],
      child: const MainWidget(),
    ),
  );
}
```

**Comment by mmomjian** [comment]
This is not something that we plan to add.
